diff --git a/src/Monolog/Formatter/LineFormatter.php b/src/Monolog/Formatter/LineFormatter.php
index e6e78983..81f443c0 100644
--- a/src/Monolog/Formatter/LineFormatter.php
+++ b/src/Monolog/Formatter/LineFormatter.php
@@ -23,7 +23,12 @@ use Monolog\Utils;
  */
 class LineFormatter extends NormalizerFormatter
 {
-    public const SIMPLE_FORMAT = "[%datetime%] %channel%.%level_name%: %message% %context% %extra%\n";
+    // This is the format used in log_stdout contrib module which works fine with Sumologic, adding it here for reference.
+    //$fmt = '[@severity] [@type] [@date] @message | uid: @uid | request-uri: @request_uri | refer: @referer | ip:  @ip | link: @link';
+    // Here's how the full output would look like in log_stdout.
+    //$output = $fmt . "\r\n")
+    // [DRUPAL LOG ENTRY] is appended for debugging purposes to ease the filtering of application logs in Sumologic
+    public const SIMPLE_FORMAT = "[%level_name%] [%channel%] [%datetime%] %message% | [DRUPAL LOG ENTRY] \r\n";
 
     /** @var string */
     protected $format;
